toonz/sources/toonz/xsheetcmd.cpp

```diff
@@ -1687,6 +1687,338 @@ class ResetArrowCommand final : public MenuItemHandler {
 
 } ResetArrowCommand;
 
+//-----------------------------------------------------------------------------
+
+class UseLinearInterpolation final : public MenuItemHandler {
```

Comment:
It seems that the classes `UseXXXXXInterpolation` are almost the same, except that the command id and the interpolation types are different.
How about unify them like as follows for simplification? : 

```
// Unify commands for all types of interpolation
class SetInterpolation final : public MenuItemHandler {
  TDoubleKeyframe::Type m_type;
public:
  SetInterpolation(CommandId cmdId, TDoubleKeyframe::Type type) :
    MenuItemHandler(cmdId), m_type(type) {}

  void execute() override {
    TApp *app = TApp::instance();
    TXsheet *xsh = app->getCurrentXsheet()->getXsheet();
    int row = app->getCurrentFrame()->getFrame();

    TStageObjectId objectId = app->getCurrentObject()->getObjectId();
    TStageObject *pegbar = xsh->getStageObject(objectId);
    if (!pegbar) return;

    int r0, r1;
    double ease0, ease1;

    pegbar->getKeyframeSpan(row, r0, ease0, r1, ease1);

    KeyFrameHandleCommandUndo *undo =
      new KeyFrameHandleCommandUndo(objectId, r0, r1);

    TStageObject::Keyframe k0 = pegbar->getKeyframe(r0);
    TStageObject::Keyframe k1 = pegbar->getKeyframe(r1);

    for (int i = 0; i < TStageObject::T_ChannelCount; i++) {
      k0.m_channels[i].m_type = m_type;
      k1.m_channels[i].m_prevType = m_type;
    }
    pegbar->setKeyframeWithoutUndo(r0, k0);
    pegbar->setKeyframeWithoutUndo(r1, k1);

    TUndoManager::manager()->add(undo);

    TApp::instance()->getCurrentScene()->setDirtyFlag(true);
    TApp::instance()->getCurrentObject()->notifyObjectIdChanged(false);
  }

} UseLinearInterpolation(MI_UseLinearInterpolation, TDoubleKeyframe::Linear)
, UseSpeedInOutInterpolation(MI_UseSpeedInOutInterpolation, TDoubleKeyframe::SpeedInOut)
, UseEaseInOutInterpolation(MI_UseEaseInOutInterpolation, TDoubleKeyframe::EaseInOut)
, UseEaseInOutPctInterpolation(MI_UseEaseInOutPctInterpolation, TDoubleKeyframe::EaseInOutPercentage)
, UseExponentialInterpolation(MI_UseExponentialInterpolation, TDoubleKeyframe::Exponential)
, UseExpressionInterpolation(MI_UseExpressionInterpolation, TDoubleKeyframe::Expression)
, UseFileInterpolation(MI_UseFileInterpolation, TDoubleKeyframe::File)
, UseConstantInterpolation(MI_UseConstantInterpolation, TDoubleKeyframe::Constant);
```

Reply:
Done.  Thanks for providing the replacement logic.